Arelle/Arelle
# Implement built-in cache for effectively permanent files

#### Reason for change
Closes #793 
Users using offline mode have to populate their cache ahead of time with common w3.org or xbrl.org files. As a matter of convenience, we can include these directly in the application since we don't expect them to change.

#### Description of change
Refactoring of the `WebCache.getfilename` function for improved readability.
Addition of a step that uses a built-in cache at `/arelle/resources/cache` if a corresponding file exists there.

#### Steps to Test
Run Arelle in offline mode and confirm operations requiring built-in cache files use them successfully.
Run Arelle in online mode and confirm operations requiring built-in cache files do not download them into the system cache and instead use the built-in cache.

**review**:
@Arelle/arelle
